### PR TITLE
Edited X2DLCInfo hook to accommodate ItemState for Issue #114 and Issue #127

### PR DESCRIPTION
**Pull request by RealityMachina**
Lets mods look at the gamestate of the Item when deciding whether to disable it from being equipped by the player. For Issue #114

### CODE REVIEW

**Comment by robojumper** [comment]
Have you tested this? What if a mod's subclass implements the function without the parameter?

**Comment by RealityMachina** [comment]
...

Right, that completely slipped my mind.

Ok, on to take two: make a wrapper function for the wrapper function. :v

File: X2WOTCCommunityHighlander/Src/XComGame/Classes/X2DownloadableContentInfo.uc [hunk 1]
```diff
@@ -273,14 +273,26 @@ static function bool GetValidFloorSpawnLocations(out array<Vector> FloorPoints,
...
+/// start Issue #114: added XComGameState_Item as something that can be passed down for disabled reason purposes
+/// basically the inventory hook wtih an added paramter to pass through
+/// we leave the old one alone for compatibility reasons, as we call it through here for those mods.
+///
+static function bool CanAddItemToInventory_CH_Improved(out int bCanAddItem, const EInventorySlot Slot, const X2ItemTemplate ItemTemplate, int Quantity, XComGameState_Unit UnitState, optional XComGameState CheckGameState, optional out string DisabledReason, optional XComGameState_Item ItemState)
+{
+	if(CheckGameState == none)
+		return CanAddItemToInventory_CH(bCanAddItem, Slot, ItemTemplate, Quantity, UnitState, CheckGameState, DisabledReason) //this can be handled by this function by newer mods
+
+	return CanAddItemToInventory(bCanAddItem, Slot, ItemTemplate, Quantity, UnitState, CheckGameState);
```
Comment (robojumper) [inline X2WOTCCommunityHighlander/Src/XComGame/Classes/X2DownloadableContentInfo.uc]: Any reason to add the check and both functions here? If `CheckGameState == none`, `CanAddItemToInventory_CH` defaults to false, and if it's not none, it calls `CanAddItemToInventory` anyway, so you should be fine replacing the function body with a call to `CanAddItemToInventory_CH`

**Comment by MalucoMarinero** [comment]
@RealityMachina just confirming this is good to go before I merge it in? Would like to cut a release in the next couple days.

**Comment by RealityMachina** [comment]
Yeah, it should be good to go.

**Pull request merged**